types/luxon/src/datetime.d.ts

```diff
@@ -146,6 +147,19 @@ export interface DateObjectUnits {
 
 export type ConversionAccuracy = 'casual' | 'longterm';
 
+export type DateTimeFormatPresetValue = 'numeric' | 'short' | 'long';
+export interface DateTimeFormatPreset {
+    year?: DateTimeFormatPresetValue;
+    month?: DateTimeFormatPresetValue;
+    day?: DateTimeFormatPresetValue;
+    weekday?: DateTimeFormatPresetValue;
+    hour?: DateTimeFormatPresetValue;
+    minute?: DateTimeFormatPresetValue;
+    second?: DateTimeFormatPresetValue;
+    timeZoneName?: DateTimeFormatPresetValue;
+    hourCycle?: 'h23';
+}
```

Comment:
This type is inherently broken.
The type allows:
```ts
DateTime.fromISO("2021-09-13T07:52:27.697Z").toLocaleString({
  ...DateTime.DATETIME_MED,
  hour: "short"
})
```
But crashes at runtime. `DateTimeFormatPresetValue` does not apply for all options in `DateTimeFormatPreset`.

Reply:
It is not restricted enough in some cases and too restrictive in other.
For example the following throws a type error but works correctly at runtime:
```js
DateTime.fromISO("2021-09-13T07:52:27.697Z").toLocaleString({
  ...DateTime.DATETIME_MED,
  hour: "2-digit"
})
```

Reply:
to be eventually fixed here: #55750